### ISSUE TITLE
WiiFlow fails to load games on vWii

### ISSUE THREAD

**Issue by AdamCatalyst** [Fledge68/WiiFlow_Lite#192]
[sysCheck.csv.zip](https://github.com/Fledge68/WiiFlow_Lite/files/4969890/sysCheck.csv.zip)

**Hello there,**

I know that variants of this issue (**WiiFlow fails to load games on vWii**) have been reported before. I have read these threads without finding resolve, and I am hoping that I can add something of value here. I have run hundreds of tests, and this is what I can report.

### Background
The reason I embarked on this insanity, is because the latest version of WiiFlow seems to crash on loading Wii games most of the time (while loading any other type of game perfectly), and it is driving me up the fracking wall. After reading about other people proclaiming the virtues of using previous versions of WiiFlow to load games, I decided to put this to try and find the best version for myself.

**Hardware Setup**
Wii U 32 GB, with a presumably a very typical vWii setup:
-  d2x-v10-beta52-vWii, base 56, cIOS slot 249
- d2x-v10-beta52-vWii, base 57, cIOS slot 250
- d2x-v10-beta52-vWii, base 58, cIOS slot 251

WiiFlow was run from a SanDisk SDXC 200 GB.

### Storage Media Tested
All media verified as authentic, and passed basic hardware tests. Furthermore, all media had been tested on the Wii U working for loading Wii U games with a 100% successful rate.
- SanDisk SDXC 200 GB
- SanDisk 3.1 USB 128 GB
- SanDisk 2.0 USB 32 GB
- Seagate Mobile HDD 160 GB (in external powered enclosure)
- Hitachi Travelstar HDD 80 GB (in external powered enclosure)

All five drives formatted as FAT32 MBR 32k clusters in Disk Utility. All WBFS files were verified pre and post copying with WIT. Volume integrity was periodically verified with Disk Utility. Volume structure was repaired, and files replaced whenever verification failed.

### WiiFlow versions tested
All test installations were completely clean, everything WiiFlow related was deleted including the WiiFLow save game file before testing a different version of WiiFlow.
- 5.2.0
- 5.4.0
- 5.4.3
- 5.4.3r35
- 5.4.6

### WiiFlow Settings Tested
All other settings were left at their default settings, except as necessary to change the partition for WiiFlow from SD to USB.
1. Force cIOS Load: **NO** / cIOS Revision: **Auto**
2. Force cIOS Load: **NO** / cIOS Revision: **249**
3. Force cIOS Load: **NO** / cIOS Revision: **250**
4. Force cIOS Load: **NO** / cIOS Revision: **251**
5. Force cIOS Load: **YES** / cIOS Revision: **Auto**
6. Force cIOS Load: **YES** / cIOS Revision: **249**
7. Force cIOS Load: **YES** / cIOS Revision: **250**
8. Force cIOS Load: **YES** / cIOS Revision: **251**

### Method
Each of the eight settings combinations was tried a minimum of twice on each of the five drives, for a minimum of 16 tests per drive with each of the five tested versions of WiiFlow. If a settings combination yielded positive results, it was given further tests. Tests were abandoned whenever a combination did not seem to be yielding results meaningfully more successful than 50%. In all, a few hundred tests were performed. WiiFlow was restarted for every single test. WiiFlow start configurator was used to change settings that seemed to prevent WiiFlow from starting.

### Findings
**It is important to note that this was meant to be a systematic way to try and find the best setup for myself, and is far from scientifically rigourous.** There were certain combinations of settings that consistently did not work with certain media or WiiFlow versions. This was expected, and so these findings are excluded from this. 

5.2.0 consistently loaded nearly every game without crashing far more than any other tested version, which really fracking sucks, because WiiFlow keeps getting better in a myriad of other ways with each new version. This was the only version of WiiFlow that loaded games ~99% of the time across any of the tested media. 

Most other versions of WiiFlow had a combination of settings that worked ~50% of the time. It is possible that some combination worked more or less than this, but this would have taken thousands of tests to discern for sure, and I nearly lost my mind doing hundreds of tests alone.

**Focusing on these inconsistent results, the overwhelming majority of crashes happened like this…**

1. Game launch was initiated from within WiiFlow.
2. Animated WiiFlow logo would appear.
3. While the logo was animating, it would momentarily visibly glitch, with what often appeared to be magenta vertical lines over the logo.
4. When the logo finished animating and disappeared, the screen would be black, and loading bar would never appear.
5. The only way to proceed was to pull the power on the Wii U.

**Lastly, these were the most successful settings combinations for each version.**

**5.2.0**
**~98%** success rate using **any media** / Force cIOS Load: **YES** / cIOS Revision: **249**
**100%** success rate using **any media** / Force cIOS Load: **YES** / cIOS Revision: **250/251**

**5.4.0**
**~75%** success rate using **SDXC** / Force cIOS Load: **NO** / cIOS Revision: **250/251** 

**5.4.3**
**~70%** success rate using **SDXC** / Force cIOS Load: **NO** / cIOS Revision: **250/251**

**5.4.3r35**
**~50%** success rate using **SDXC** / Force cIOS Load: **NO** / cIOS Revision: **250/251**

**5.4.6**
**~50%** success rate using **any USB** / Force cIOS Load: **NO** / cIOS Revision: **250**
**~50%** success rate using **SDXC** / Force cIOS Load: **NO** / cIOS Revision: **251**

### Conclusion
Clearly there is an issue here, but it is not so clear what it is…
- I have seen so little variation in the drives that I tested, that I believe I can rule out it being a problem with my hardware.
- That being said, all the drives were formatted the same way, which could be the issue. I will explore other formatting methods next. 
- Obviously it could be something with how my vWii is setup. To that end I have included my SysCheck report and ModMii's opinion about it below. 
- Regardless, it seems like something changed in 5.4.0 that reduced WiiFlow's flexibility for whatever my underlying issue is
- It also seems like something changed in 5.4.3r25 that further reduced WiiFlow's flexibility for whatever my underlying issue is

I am grateful to discover that I can use WiiFlow 5.2.0 and have my games launch virtually perfectly. I am also grateful that you continue to develop and improve on WiiFlow. I wish I (and others who are having a similar issue) could be able to enjoy the refinements in WiiFlow's development past 5.2.0 without it's key feature working inconsistently for us.

**I would be happy to answer any questions to try and facilitate understanding this issue better.**

p.s. According to ModMii:

> Your unique list of WADs to install are as follows: 
> IOS11v16174(IOS60v6174[FS-ES-NP-VP-DIP])
> IOS20v16174(IOS60v6174[FS-ES-NP-VP-DIP])
> IOS30v16174(IOS60v6174[FS-ES-NP-VP-DIP])
> IOS40v16174(IOS60v6174[FS-ES-NP-VP-DIP])
> IOS50v16174(IOS60v6174[FS-ES-NP-VP-DIP])
> IOS52v16174(IOS60v6174[FS-ES-NP-VP-DIP])
> IOS60v16174(IOS60v6174[FS-ES-NP-VP-DIP])
> IOS70v16174(IOS60v6174[FS-ES-NP-VP-DIP])
> IOS80v16174(IOS60v6174[FS-ES-NP-VP-DIP])
> cIOS202[60]-v5.1R
> cIOS222[38]-v4
> cIOS223[37-38]-v4
> cIOS224[57]-v5.1R
> cIOS249[57]-d2x-v8-final
> cIOS250[56]-d2x-v8-final
> RVL-mios-v10
> IOS9-64-v1034
> IOS12-64-v526
> IOS13-64-v1032
> IOS14-64-v1032
> IOS15-64-v1032
> IOS17-64-v1032
> IOS21-64-v1039
> IOS22-64-v1294
> IOS28-64-v1807
> IOS31-64-v3608
> IOS33-64-v3608
> IOS34-64-v3608
> IOS35-64-v3608
> IOS36-64-v3608
> IOS37-64-v5663
> IOS38-64-v4124
> IOS41-64-v3607
> IOS43-64-v3607
> IOS45-64-v3607
> IOS46-64-v3607
> IOS48-64-v4124
> IOS53-64-v5663
> IOS55-64-v5663
> IOS56-64-v5662
> IOS57-64-v5919
> IOS58-64-v6176
> IOS61-64-v5662
> IOS62-64-v6430

**Comment by gingerbeardman** [comment]
This is awesome! Thanks sincerely for going to this effort. Let me begin to digest the contents.

**Comment by unresolvedsymbol** [comment]
WiiFlow has a 100% success rate for my vWii, the only thing I have to do is power off my Wii U formatted USB HDD beforehand... Your tests don't note if you had a Wii U formatted drive connected

**Comment by AdamCatalyst** [comment]
When running the tests, all peripherals that were not necessary for the respective test were disconnected. The SD card was, however, a constant, as WiiFlow is installed on it. I only mention it because any constant could be the source of my woes.

One potential problem that I did mention is that all my drives were formatted using Apple's Disk Utility (latest version). I did some quick tests with Bootice & GUIformat, and in both cases the volumes could not be read on my vWii. I was booted in Boot Camp (as I unfortunately only have Macs to test with) which I suppose could be the reason for this. Many people use those formatters, so clearly they work, and yes, I know how to format a disk properly :)

As an aside, I have no problem with using SD, or USB in WiiFlow 5.2.0 even with my Wii U hard drive connected and awake. I've had 100% stability since the downgrade, regardless of the amount of peripherals connected, or whatever other crazy nonsense I get up to.

**Comment by Danster64** [comment]
Oh neat, I was having trouble with WiiFlow Lite when I upgraded to it today. I'm also trying to run Wii games on my vWii off a FAT32 64 GB SD Card. It's worked fine on an older version of WiiFlow before. I've set my cIOS' match Adam's and there aren't any Wii U formatted drives connected.
I'll downgrade to v5.2.0 for now and keep posted for any updates that try and fix this~

**Comment by Fledge68** [comment]
for all those vwii users who say wiiflow v5.2.0 works and any version 5.4.0 onward they have issues.
here is a folder of wiiflow beta commits between version 5.2.0 and 5.40. each one includes the boot.dol and ext_booter.bin. the booter.bin goes in the bins folder of course.
I suggest you download and use 5.2.0 release first. then simply download and plug the boot dol and booter bin to the appropriate folders and try again. be sure to backup your wiiflow stuff before hand.
let me know which downloads work and which don't. maybe this way we can finally figure out what commit between 5.2.0 and 5.4.0 messed things up.

here's the downloads - https://www.mediafire.com/folder/vldp2pufn1r3l/

**Comment by AdamCatalyst** [comment]
AWESOME. I will run the tests as you describe and report. Thanks for prepping this!

**Comment by Fledge68** [comment]
any progress? sorry i get a little impatient sometimes.

**Comment by AdamCatalyst** [comment]
I did spend some time running tests yesterday, and thought I had figured it out, but when I went to verify my results through replication, replication failed to produce the same results. I will re-do the tests, and get rid of some possible confounders. Sorry, it will probably take me a few days, as I am crazy busy today and tomorrow. I don't want to point to any commit until I am certain I am not wasting your time by sending you on a wild good chase.

**Comment by AdamCatalyst** [comment]
### _tldr;_ **2019 10 23 commit 5.3.0 Beta 12 (TLS Support)**

Going to be quick, as I'm in a rush, but happy to answer questions if you have any. In brief…

**2019 05 27 Beta**
- Force cIOS Load: **NO** / cIOS Revision: 250 = **0% (0/1)** successful loads (the only test resulted in what appeared to be the exact error described in my original post)
- Force cIOS Load: **YES** / cIOS Revision: 250 = **100% (22/22)** successful loads.

**2019 09 23 Beta**
- Force cIOS Load: **YES** / cIOS Revision: 250 = **100% (18/18)** successful loads.

**2019 10 23 Beta**
- Force cIOS Load: **NO** / cIOS Revision: 250 = **78%–61% (11/14 to 11/18)** successful loads.* (errors were all consistent with the glitch as described in my original post)
- Force cIOS Load: **YES** / cIOS Revision: 250 = **40% (2/5)** successful loads. (errors were just black screens, and not consistent with the glitch as described in my original post)

**Q:** _Why didn't you test every Beta?_
Crunched for time. Used split-half method of testing.

**Q:** _Why are you reporting a range of numbers for the 2019 10 23 Beta?_
After four consecutive failed tests of the same games, I removed the SD card to check for FAT corruption and verify the games that failed to load. While the games verified fine, the partition had developed FAT corruption. It is therefore possible that this was the underlying cause of those four failed load attempts.

**Q:** _Why didn't you test these Betas the same amount of times?_
After the corruption discover, from then on, when games would fail, I would switch back to the 2019 05 27 Beta to verify that the specific game was still loading fine.

**Q:** _How do we know that other errors weren't simply caused by directory corruption?_
We don't know for sure, since I did not test the partition and games after every single failed load. However, at the end of testing, I scanned the card and tested the games again. No errors. That also means my results pre and post testing the partition and games integrity was virtually identical.

**Comment by Fledge68** [comment]
so just to clarify - i see you refer to 10-23-2019 which is actually the last commit BEFORE tls support, right?

if that's the case then we need to look at the commits between 9-23 and 10-23.
but a quick glance shows that most are for the newly added snapshots and none of them seem to have anything to do with booting a game.

I'm also curious as to when sd card was corrupted. if 10-23 did it or if it was actually before 10-23.

**Comment by AdamCatalyst** [comment]
Ahhh yes. I minced words, and will correct my post above. To be clear, the version you gave me labeled **oct 24 & 26 2019.zip** is the one that gives me the error. 

Unfortunately, I have no way of saying for sure what version caused the FAT corruption issues. I have found on my Wii and Wii U that FAT corruption happens from time to time if the system freezes on some read/write operations. I have never thought it to be something WiiFlow was doing, but it became evident it was happening in WiiFlow when I did the initial test at the top of this thread. My anecdotal observation was that the corruption was caused by crashes, and didn't precede it. I personally find FAT 32 to go corrupt at a drop of a hat.  

I'm going to pause here on testing. Both the error, and the frequency of the error, are inline with the other tests I've done, so I don't feel the need to test this further. But if you have any requests, or more interim builds, I am happy to do more testing for you.

**Comment by Fledge68** [comment]
hmm, ok.

i was just checking the code about the snapshot. I'm thinking the m_snap and m_overlay are not being cleared when launching a game. they are when you exit the game selected menu and if no image is found but not when booting a game. there is theme cleanup but it only deletes texture sets which are used for button's. as far as i can non clickable images such as snapshots are never added to a texture set and thus never cleared during cleanup.

if the tls support is really the problem. the immediate thing i think of is i wonder if its' de initializing or shutting down properly on wii u vwii during game boot. I'll see if Blackb0x has any thoughts

**Comment by Fledge68** [comment]
one more thing. since you now say it wasn't 10-23 commit and that it was 10-24 & 10-26 commits. so did you try 10-23? because if not then that means it could very well be 10-23 before tls support was added.

I realize you are busy ( we all are) so whenever you can find time to fix your corrupted device and try 10-23 i'd appreciate it.  but hopefully some time before the end of the month would be nice.

 or if anyone else with this similar problem wants to test.

**Comment by AdamCatalyst** [comment]
Actually, in the archive you shared, I did not see any 10-23 builds, only 9-23, and one labelled 10-24 & 26. Send me 10-23 (or point out where I missed it) and I'll happily test it ASAP! (within days at the latest)

**Comment by Fledge68** [comment]
ok thanks.

10-23 - http://www.mediafire.com/file/pu1xf370gkwur3v/file

**Comment by AdamCatalyst** [comment]
**2019 10 23 Beta**

Force cIOS Load: **NO** / cIOS Revision: 250 = **0% (0/1)** successful loads (the only test resulted in what appeared to be the exact error described in my original post)
- Force cIOS Load: **YES** / cIOS Revision: 250 = **100% (18/18)** successful loads.

**Comment by AdamCatalyst** [comment]
Just checking in to make sure you got my notes on the 10-23 build. I'm happy to test anything anytime!

**Comment by Fledge68** [comment]
@AdamCatalyst yes i saw your test results. but i can't imagine adding wolfssl to wiiflow would affect game launching on wii u vwii. mainly because its not even used or initialized unless you download covers. therefore the only thing it does is make wiiflow larger in bytes.

**Comment by wiidev** [comment]
@AdamCatalyst Could you please test these versions? They're all based on the latest code.

https://we.tl/t-6RcOUfpvgT

If you see a crash dump please get a good picture of it and tell me which version it belongs to.

**Comment by AdamCatalyst** [comment]
@AdamCatalyst I hear ya.
@wiidev I am on it. Give me 12-24 hours.

**Comment by wiidev** [comment]
@AdamCatalyst To speed up your tests I'd suggest starting with the normal version first. Only if it fails will you need to try the other two.

- The normal version is built against the latest code
- The test version removes a wolfSSL method that's called before booting games
- The no SSL version removes the wolfSSL lib entirely

If all 3 of them have issues then we'll know that the issues not caused by wolfSSL.

**Comment by AdamCatalyst** [comment]
**Normal**
Force cIOS Load: NO / cIOS Revision: 250 = 64% (9/14) successful loads.

**Test**
Force cIOS Load: NO / cIOS Revision: 250 = 71% (17/24) successful loads.

**No SSL**
Force cIOS Load: NO / cIOS Revision: 250 = 65% (15/23) successful loads.

Not seeing any statistically significant difference between these three variants.

In my initial tests, results could be grouped as builds that performed, 100% of the time (5.2.0), builds that loaded ~70% (5.4.0–5.4.3) of the time, and builds that loaded 50% (5.4.3r35–5.4.6) of the time. With the amount of tests I've performed, I can say with confidence that this build has a statistically significant higher likelihood to load games on a problematic vWii like mine, than builds 5.4.3r35–5.4.6. It is the most stable version I have tested since 5.4.3. *Something* is working better for me, even if the bigger issue remains.

Please feel free to send me more builds, even old ones that you have mislabeled. You have my full consent to lie to me about what you are sending me. :)

**Comment by wiidev** [comment]
Based on them results I think it's safe to say that the issue isn't caused by wolfSSL. So the issues started before I ever contributed code to WiiFlow.

In your original message you said this...
> **5.2.0**
> **~98%** success rate using **any media** / Force cIOS Load: **YES** / cIOS Revision: **249**
> **100%** success rate using **any media** / Force cIOS Load: **YES** / cIOS Revision: **250/251**

But what about 5.2.0 without force loading the cIOS?

Looking at the commit history I can see that 5.2.0 originally would boot with IOS58, but [a change was made for 5.4.0](https://github.com/Fledge68/WiiFlow_Lite/commit/7bf04b023d22978ead69240ca60eefbea42d1139) and it now uses cIOS 249.

**Comment by Fledge68** [comment]
in the previous message he says force cios ON 100%

2019 10 23 Beta

    Force cIOS Load: NO / cIOS Revision: 250 = 0% (0/1) successful loads (the only test resulted in what appeared to be the exact error described in my original post)
    Force cIOS Load: YES / cIOS Revision: 250 = 100% (18/18) successful loads.

why did he choose to do force cios OFF for these tests? if force cios ON is working why not just use it?

**Comment by AdamCatalyst** [comment]
> But what about 5.2.0 without force loading the cIOS?

Sadly, I have lost my paper notes that I made while testing. I cannot tell you how many time it worked or didn't work, but it was tested. As my goal was to find a combination that worked for me, I stopped testing any setting that gave me repeated errors, when an alternate setting appeared like it might work 100% of the time. I don't recall how far or close the OFF setting was in that case. Do you want me to test that combination and report back?

**Comment by AdamCatalyst** [comment]
> why did he choose to do force cios OFF for these tests? if force cios ON is working why not just use it?

Ahh, because somewhere between 5.2.0 and 5.4.0 something happened, and subsequent versions of WiiFlow either would not start or would only load games sporadically (if at all) when Force cIOS was set to ON for me. If you want me to go back and test a specific setting combination that I deemed not working, just let me know how many flips of the coin you want to be sure.

**Comment by wiidev** [comment]
> Do you want me to test that combination and report back?

Well it might be good to know how well the 3 versions work that I gave you if you force the cIOS. We could then compare it to the 2019/10/24 build.

**Comment by AdamCatalyst** [comment]
@wiidev OK, should be able to squeeze that in today.

**Comment by AdamCatalyst** [comment]
OK, supplementary test…

**Normal**
Force cIOS Load: YES / cIOS Revision: 250 = 0% (0/4) successful loads.

**Test**
Force cIOS Load: YES / cIOS Revision: 250 = 0% (0/3) successful loads.

**No SSL**
Force cIOS Load: YES / cIOS Revision: 250 = 0% (0/2) successful loads.

I know these are small sample sizes, but it looks pretty clear like it won't load. With the other error, it got through the pre-loader graphic, and when it should jump to the white loading bar, it just goes black. When Force cIOS is set to YES, it only shows the pre-loader graphic for a tiny fraction of a second before the system locks up. No error screens in either case.

**Comment by wiidev** [comment]
@AdamCatalyst Hey, I've got some builds that you can test if you're willing?

https://we.tl/t-dyR51NMZll

We want you to do the tests with cIOS 249/250 force loaded, like you did with WiiFlow 5.2.0 to get the 100% success rate.

You should probably backup your wiiflow_lite.ini file and delete the new file that's created after each test.

**Comment by wiidev** [comment]
@AdamCatalyst Hey, I've got some builds that you can test if you're willing?

https://we.tl/t-Bz4EjIwCw6

We want you to do the tests with cIOS 249/250 force loaded, like you did with WiiFlow 5.2.0 to get the 100% success rate.

You should probably backup your wiiflow_lite.ini file and delete the new file that's created after each test.

**Comment by AdamCatalyst** [comment]
@wiidev Will do!

**Comment by AdamCatalyst** [comment]
I have started testing. All tests will be done with:

**Force cIOS Load: YES / cIOS Revision: 250**

All WiiFlow related files will be deleted after finding testing a build. I will proceed to 16 positive tests, or 4 failed tests, whichever occurs first. The only custom setting used will be returning to the WiiFlow channel, to speed up testing. I am purposely avoiding looking at any details of the builds, so I have no clue what I am testing, on the number at the end of the folder name.

I am going to post results here after I finish testing a build to the criteria above, in case that result is enough for you to say _"Hey, you can stop now, and quit procrastinating from your work!"_. Please do let me know that. :) Without further ado…

**5fb7a5a_8** = 100% (16/16)

**Comment by wiidev** [comment]
@AdamCatalyst You can stop there since _6 is older than _1.

There's only a few more commits until I get to the v5.4.0 release build, which you previously had issues with. So I'll compile them and we'll see what happens.

**Comment by wiidev** [comment]
@AdamCatalyst You can stop there since _6 is newer than _1.

There's only a few more commits until I get to the v5.4.0 release build, which you previously had issues with. So I'll compile them and we'll see what happens.

**Comment by wiidev** [comment]
https://we.tl/t-VdY9XNZUqL

_21 is basically v5.4.0 compiled from my PC, so you'll probably want to start with that and work your way down until you're getting consistent successful loads.

**Comment by wiidev** [comment]
https://we.tl/t-PLpoYmKcaN

_21 is basically v5.4.0 compiled from my PC, so you'll probably want to start with that and work your way down until you're getting consistent successful loads.

**Comment by AdamCatalyst** [comment]
@wiidev Will do, but gonna use "split-half" method if these are sequential iterative builds and you are expecting a single fail point. In that case split-half is arguably more time efficient, and therefore less likely to lead to divorce.

**Comment by AdamCatalyst** [comment]
@wiidev OK, that proved to be easy.
**_14** = 100% (8/8)
**>_14** = 0%, but not the error I was initially reporting. It is the error I reported in https://github.com/Fledge68/WiiFlow_Lite/issues/192#issuecomment-690617247 which occurs when cIOS settings are not working. Do you want me to stick with cIOS 249/250, force load ON?

**Comment by wiidev** [comment]
@AdamCatalyst Yes, but as it fails on _15 or newer you can skip the other tests.

Try this > https://we.tl/t-j5lPNvzoHM

**Comment by wiidev** [comment]
@AdamCatalyst Yes, stick to using cIOS 250 with force load turned on. But as it fails on _15 or newer you can skip the other tests.

Try this https://we.tl/t-j5lPNvzoHM

**Comment by AdamCatalyst** [comment]
**Test1** = 94% (15/16)

The only failed load was completely atypical, and **not** the glitch I described in my initial post. I'm continuing to test this well past the 16 load mark, literally just running through all my games. Nothing is failing.

**Comment by wiidev** [comment]
As it's also built against the latest version of the code it looks like I've found the issue. 

If you were to do the exact same tests with v5.4.0 the success rate should be similar.

**Comment by AdamCatalyst** [comment]
I can't tell you how excited I am. Can I ask, what the issue was?

**Comment by wiidev** [comment]
If you take a look at [this commit](https://github.com/Fledge68/WiiFlow_Lite/commit/cc715cf04acd4efea8d26e6480e33162463447a4#diff-b3634f98cbfda72360afbb20e822525b) you can see that net_wc24cleanup() was moved into the ShutdownBeforeExit() function, since that seemed like the logical thing to do. But when you boot a game it calls ShutdownBeforeExit() and apparently vWii doesn't like it when the net_wc24cleanup() function is used before starting a game.

**Comment by AdamCatalyst** [comment]
Thank-you for explaining!

**Comment by AdamCatalyst** [comment]
One other anecdotal observation. Doing all these tests I noticed that it was possible to crash WiiFlow if I took a brand new install, and launched a game while WiiFlow was still trying to download the banner for that game. I can't quantify this, as I never kept records, it is just a pattern I started to notice recently, and why I dismissed that one fail one the test build you sent me. Somehow I doubt very many people are loading up a fresh install of WiiFlow and trying to load a game the instant WiiFlow opens. And its not exactly rare for any software to glitch if you button mash fast enough.

**Comment by wiidev** [comment]
And thank you for testing! I've sent @Fledge68 a pull request.

> I noticed that it was possible to crash WiiFlow if I took a brand new install, and launched a game while WiiFlow was still trying to download the banner for that game.

I don't own a Wii U to test it myself, but with a Wii you can't start a game while it's downloading as you can't exit the download screen until it's complete. And it does that because the sockets need to be closed and the data either needs to be written to disk or released.

**Comment by wiidev** [comment]
@AdamCatalyst Could you confirm if this version also works?

https://we.tl/t-sB8ZlSglMj

I adjusted the code again because like I said originally it made sense for me to relocate net_wc24cleanup() into the ShutdownBeforeExit() function, since that free's everything up correctly. So I believe the issue is actually caused by net_wc24cleanup() getting called before net_deinit().

Thanks.

**Comment by wiidev** [comment]
@AdamCatalyst Could you confirm if either of these work for you too?

https://we.tl/t-zJVZXtXnLX

I adjusted the code again because I suspect the issue might actually be caused by net_wc24cleanup() getting called before net_deinit().

Thanks.

**Comment by wiidev** [comment]
@AdamCatalyst Could you confirm if either of these work for you too?

https://we.tl/t-TQvnO0Wczl

I adjusted the code again because I suspect the issue might actually be caused by net_wc24cleanup() getting called before net_deinit().

Thanks.

**Comment by AdamCatalyst** [comment]
@wiidev Will do, tomorrow afternoon EST.

**Comment by gingerbeardman** [comment]
Thanks very much to both of you @AdamCatalyst @wiidev !

**Comment by AdamCatalyst** [comment]
@wiidev The second version, failed in 2/2 tests. Can do more if you need it, was just pressed for time right now. The third version loaded 16/16 times, no issues.

**Comment by wiidev** [comment]
@AdamCatalyst That's actually an interesting result, since it means that networkInit must be false for you 🤔

I guess as long as test 3 passes then we should be good to go. So I'll update my pull request and @Fledge68 will probably release an update when he's got some free time.
